lib/stripe/stripe_client.rb

```diff
@@ -120,36 +120,38 @@ def execute_request(method, path,
       params = Util.objects_to_ids(params)
       url = api_url(path, api_base)
 
+      body = nil
+      query_string = nil
+
       case method.to_s.downcase.to_sym
       when :get, :head, :delete
-        # Make params into GET parameters
-        url += "#{URI.parse(url).query ? '&' : '?'}#{Util.encode_parameters(params)}" if params && params.any?
-        payload = nil
+        query_string = Util.encode_parameters(params) if params && params.any?
+        url += "#{URI.parse(url).query ? '&' : '?'}#{query_string}" unless query_string.nil?
       else
-        payload = if headers[:content_type] && headers[:content_type] == "multipart/form-data"
-                    params
-                  else
-                    Util.encode_parameters(params)
-                  end
+        body = if headers[:content_type] && headers[:content_type] == "multipart/form-data"
```

Comment:
For clarity, I renamed `payload` to `body` everywhere. `payload`'s relatively obvious, but `body` is more clear and less ambiguous.